src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +378,313 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
+            int result = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+
+                if (v2Region.StartLine > 0) // Use line and column 
+                {                    
+                    string sourceLine = null;
+
+                    // Read down to the line before startLine
+                    for (int i = 1; i < v2Region.StartLine; sourceLine = reader.ReadLine(), i++) { }
+
+                    // Read to startColumn
+                    char[] buffer = new char[v2Region.StartColumn];
+                    reader.Read(buffer, 0, buffer.Length);
+
+                    // Read the next charLength characters
+                    buffer = new char[v2Region.CharLength];
+                    reader.Read(buffer, 0, buffer.Length);
+
+                    result = encoding.GetByteCount(buffer);
+                }
+                else // Use charOffset
+                {
+                    // Read the next charLength characters
+                    char[] buffer = new char[v2Region.CharOffset];
+                    reader.Read(buffer, 0, buffer.Length);
+
+                    result = encoding.GetByteCount(buffer);
+                }
+            }
+            finally
+            {
+                if (reader != null)
+                {
+                    reader.Dispose();
+                }
+            }
+
+            return result;
+        }
+
+        private int GetRegionEndColumn(Region v2Region, Uri uri)
+        {
+            int result = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+                string sourceLine = null;
+
+                // Read down to the line before endLine
+                for (int i = 1; i < v2Region.EndLine; sourceLine = reader.ReadLine(), i++) { }
+                result = sourceLine.Length;
+            }
+            finally
+            {
+                if (reader != null)
+                {
+                    reader.Dispose();
+                }
+            }
+
+            return result;
+        }
+
+        private TextReader GetFileTextReader(Uri uri, out Encoding encoding)
+        {
+            TextReader reader = null;
+            encoding = null;
+            var failureReason = new StringBuilder();
+
+            if (uri != null && _currentV2Run.Files != null)
+            {
+                FileData fileData;
+                if (_currentV2Run.Files.TryGetValue(uri.OriginalString, out fileData))
+                {
+                    // Determine the encoding
+                    string encodingName = null;
+
+                    if (fileData.Contents?.Text != null)
+                    {
+                        // Embedded text shall be UTF-8 encoded
+                        encoding = Encoding.UTF8;
```

Comment:
The fact that the `file` object contains a `contents.text` property, together with the fact that text in a SARIF file is always encoded in UTF-8, does _not_ mean that the file's encoding is UTF-8. The spec explains (§3.2.2, fileContents.text property):
> If the external file is a text file, a `fileContent` object **SHOULD** contain a property named `text` whose value is a string containing the relevant text. Since SARIF log files are encoded in UTF-8 ([RFC3629]; see §3.1), this means that if the external file is a text file in any encoding other than UTF-8, the SARIF producer **SHALL** transcode the text to UTF-8 before assigning it to the text property...

So you see, the contents of `file.contents.text` might have been transcoded. You have to rely on the `file.encoding` property. If it's absent (and `run.defaultFileEncoding` is missing), you're out of luck, because it has no default value (see §3.17.9, file.encoding property):
> If the `file` object represents a text file and this property is absent, it **SHALL** default to the value of the `defaultFileEncoding` property (§3.11.18) of the containing `run` object (§3.11), if that property is present; otherwise, the file’s encoding **SHALL** be taken to be unknown.

Reply:
Fixed

Reply:
Fixed